src/index.js

```diff
@@ -5,9 +5,37 @@
 const fs = require('fs');
 const feedQueue = require('./feed-queue');
 const feedWorker = require('./feed-worker');
+const parentLogger = require('../src/lib/logger');
+
+const log = parentLogger.child({ module: 'main' });
 
 // Start the web server
-require('./backend/web/server');
+const server = require('./backend/web/server');
+
+/**
+ * Stops the Redis Queue, closes all connections gracefuly
+ */
+function shutDown() {
+  log.info('Received kill signal, shutting down gracefully');
+
+  // Close the Redis Queue
+  feedQueue.close();
+
+  // Close server and terminate all connections
+  server.close(() => {
+    log.info('Closed out remaining connections');
+    process.exit(0);
+  });
+
+  // Force shutting down
+  setTimeout(() => {
+    log.error('Could not close connections in time, forcefully shutting down');
+    process.exit(1);
+  }, 10000);
+}
+
+process.on('SIGTERM', shutDown);
+process.on('SIGINT', shutDown);
```

Comment:
I wonder if we should just move all the "shutdown" logic in here.  In `src/backend/web/server.js` we have:

```js
process.on('unhandledRejection', err => {
  console.log('UNHANDLED REJECTION:  Shutting down...');
  console.log(err.name, err.message);
  server.close(() => {
    process.exit(1);
  });
});
```

Should we move that block of code here?  Then it's all contained in one place.

Reply:
It looks for me as `process.on('unhandledRejection',...` is more related to any case of error handling (like, for all of the errors), while `SIGINT` and `SIGTERM` are more related to Shutdown - so they could stay separated.

But I believe none of the logic will be broken if we move it.

Reply:
Maybe we could export `shutdown` function and call it from other modules like:
``` 
process.on('unhandledRejection', importedModule.shutDown(error));
```
that requires slightly updating the shutDown to log the error ...